### PR TITLE
[loadgen] Support workspace classes

### PR DESCRIPTION
**Pull request by Furisto**
## Description
- Select workspace classes for load testing
- Better documentation
- Better support for debugging issues found during load testing
- Adapt values for prod benchmark 

## Related Issue(s)
None

## How to test
You can actually now just read the documentation :wink: 

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: dev/loadgen/README.md [hunk 1]
```diff
@@ -2,4 +2,58 @@
...
+`loadgen benchmark [config-file] --host localhost:12001 --tls ./wsman-tls`
+
+In order to configure the benchmark, you can use the configuration file
+
+| Parameter  | Description |
+| ------------- | ------------- |
+| workspaces  | The number of workspaces that will be started during the benchmark  |
+| ideImage  | The image that will be used for the IDE |
+| waitForRunning | How long to wait for workspaces to enter running state |
+| waitForStopping | How long to wait until all workspaces are stopped |
```
[the remark this answers is not in the document]
Comment (Furisto) [reply dev/loadgen/README.md:26]: @kylos101 I updated the documentation to mention the --interactive flag and linked to a loom video explaining the tool.

[the remark this answers is not in the document]
Comment (kylos101) [reply dev/loadgen/README.md:26]: Awesome, on it!

[the remark this answers is not in the document]
Comment (kylos101) [reply dev/loadgen/README.md:26]: @aledbf I think `--interactive` will help you in the future (if you haven't already used).

File: dev/loadgen/README.md [hunk 2]
```diff
@@ -2,4 +2,60 @@
...
+| successRate | Percentage of started workspaces that should enter running state to count as a successful run
+| environment | Global environment variables that will be set for all repositories |
+| workspaceClass | The workspace class to use for workspaces. This setting can be overriden for individual repositories.
+| repos | The repositories that will be used to create workspaces |
+| repo.cloneURL | The url of the repository |
+| repo.cloneTarget | The branch to clone from |
+| repo.score | The score decides how often a repository is used for the creation of a workspace. |
+| repo.workspaceImage | The docker image that will be used for the workspace |
+| repo.environment | Environment variables that will only be set for this repository |
+| repo.workspaceClass | The workspace class to use for the workspace that will be created for this repository |
```
Comment (kylos101) [inline dev/loadgen/README.md:36]: :love_letter:

File: dev/loadgen/pkg/loadgen/executor.go [hunk 3]
```diff
@@ -112,7 +117,7 @@ type WsmanExecutor struct {
 
 // StartWorkspace starts a new workspace
 func (w *WsmanExecutor) StartWorkspace(spec *StartWorkspaceSpec) (callDuration time.Duration, err error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 240*time.Second)
```
Comment (kylos101) [inline dev/loadgen/pkg/loadgen/executor.go:120]: @aledbf wdyt of this timeout? I assume it matches your expectation, but wanted to double check.

File: dev/loadgen/prod-benchmark.yaml [hunk 4]
```diff
@@ -1,14 +1,15 @@
 ## start with
 ##    loadgen benchmark prod-benchmark.yaml
 
-workspaces: 10
+workspaces: 100
 ideImage: eu.gcr.io/gitpod-core-dev/build/ide/code:commit-ff263e14024f00d0ed78386b4417dfa6bcd4ae2f
 waitForRunning: "600s"
 waitForStopping: "600s"
-successRate: 0.80
+successRate: 0.95
```
Comment (kylos101) [inline dev/loadgen/prod-benchmark.yaml:8]: Honest question, why not 100%? 

What does that rationale or use case look like, when it is >= 95, but, < 100?

**Review by kylos101 — COMMENTED** [review]
Hey @Furisto , just the [one question](https://github.com/gitpod-io/gitpod/pull/10409#discussion_r887356687), solid work amigo! Good call sharing the Loom video, too.

**Pull request merged**